actors/evm/src/interpreter/stack.rs

```diff
@@ -10,78 +12,111 @@ const INITIAL_STACK_SIZE: usize = 32;
 /// EVM stack.
 #[derive(Clone, Debug)]
 pub struct Stack {
-    sk: Vec<U256>,
-    d: usize,
+    stack: Vec<U256>,
 }
 
 impl Stack {
     #[inline]
     pub fn new() -> Self {
-        Stack { sk: Vec::from([U256::zero(); INITIAL_STACK_SIZE]), d: 0 }
+        Stack { stack: Vec::with_capacity(INITIAL_STACK_SIZE) }
     }
 
-    #[inline]
-    pub fn require(&self, required: usize) -> bool {
-        required <= self.d
+    #[inline(always)]
+    pub fn len(&self) -> usize {
+        self.stack.len()
     }
 
-    #[inline]
-    pub fn ensure(&mut self, space: usize) -> bool {
-        let required = self.d + space;
-
-        if required > self.sk.len() {
-            if required > STACK_SIZE {
-                return false;
-            }
-
-            while required > self.sk.len() {
-                self.sk.resize(2 * self.sk.len(), U256::zero());
-            }
-        }
-        true
+    #[inline(always)]
+    pub fn is_empty(&self) -> bool {
+        self.stack.is_empty()
     }
 
     #[inline(always)]
-    pub fn len(&self) -> usize {
-        self.d
+    pub fn push_unchecked(&mut self, value: U256) {
```

Comment:
This is still very much checked, could use a better name.

But given our usage patterns it could be very much _really_ (unsafe) unchecked